### DISCUSSION TITLE
Anything look slow in this rofi blezz timing profile?

### DISCUSSION THREAD

**Discussion by focusaurus** [davatorium/rofi#1893] [Q&A]
[Here's a gist with the profile timings and the command line I use](https://gist.github.com/focusaurus/072ecd7293d097db697e3d35f826730a). Starting a new discussion from [this comment in a thread about leaving the rofi window running long term](https://github.com/davatorium/rofi/discussions/1594#discussioncomment-6825645).

I've never read one of these before but I think it's pretty much done in about 275ms which is not slow, but I can realistically type 2 keystrokes faster than that and have the second one still grabbed by the previous application.

The longest time between 2 logs I think is here:  

```
(process:185580): Timings-DEBUG: 15:05:57.342: 0.096727 (0.000015): source/view.c:rofi_view_repaint:286 flush
(process:185580): Timings-DEBUG: 15:05:57.522: 0.276428 (0.179701): source/view.c:rofi_view_refilter_real:1170 Filter start
```

**Comment by DaveDavenport** [comment]
It looks like that is when it start responding to your input, as it refilters the list. 

From this point on it should be taking your keyboard input, and it should not land in any other application anymore:
(process:185580): Timings-DEBUG: 15:05:57.271: 0.024924 (0.000007): source/rofi.c:startup:728 Grab keyboard

That is 24 ms. 

How do you launch rofi?  (what hotkey tool?)

**Comment by focusaurus** [comment]
I am running awesomewm and have a key mapped in my lua code to `awful.spawn.easy_async(home_bin .. '/blezz', noop)` which runs my blezz script which is in the gist.

**Reply by DaveDavenport** [reply]
hmm I have very little experience with awesomewm. maybe if I have some time I can take a look. (but won't be soon, as I have negative time to spend on rofi).

**Reply by focusaurus** [reply]
Don't worry about it.

**Comment by focusaurus** [comment]
Gonna hard code user id instead of running `id -u` and see if that helps.
